pkg/adapterManager/manager2.go

```diff
@@ -0,0 +1,439 @@
+func NewManager2(mapper expr.Evaluator, templateRepo template.Repository,
+	rt RuntimeState, gp *pool.GoroutinePool) Dispatcher {
+	m := &Manager2{
+		mapper:       mapper,
+		templateRepo: templateRepo,
+		gp:           gp,
+	}
+	m.SetRuntimeState(rt)
+	return m
+}
+
+// Manager2 is the 0.2 adapter manager.
+type Manager2 struct {
+	// mapper is the selector and expression evaluator.
+	// It is passed through by Manager.
+	mapper expr.Evaluator
+
+	// Repository of templates that are compiled in this Mixer.
+	// It provides a way to perform template operations.
+	templateRepo template.Repository
+
+	// <RuntimeState>
+	rtState atomic.Value
+
+	// gp is used to run manager code asynchronously.
+	gp *pool.GoroutinePool
+}
+
+// SetRuntimeState installs a new runtime state.
+func (m *Manager2) SetRuntimeState(state RuntimeState) {
+	m.rtState.Store(state)
+}
+
+// RuntimeState gets the current runtime state.
+func (m *Manager2) RuntimeState() RuntimeState {
+	v := m.rtState.Load()
+	if v == nil {
+		return nil
+	}
+	return v.(RuntimeState)
+}
+
+// Dispatcher executes aspects associated with individual API methods
+type Dispatcher interface {
+
+	// Preprocess dispatches to the set of aspects that will run before any
+	// other aspects in Mixer (aka: the Check, Report, Quota aspects).
+	Preprocess(ctx context.Context, requestBag attribute.Bag, responseBag *attribute.MutableBag) error
+
+	// Check dispatches to the set of aspects associated with the Check API method
+	Check(ctx context.Context, requestBag attribute.Bag) (*adapter.CheckResult, error)
+
+	// Report dispatches to the set of aspects associated with the Report API method
+	Report(ctx context.Context, requestBag attribute.Bag) error
+
+	// Quota dispatches to the set of aspects associated with the Quota API method
+	Quota(ctx context.Context, requestBag attribute.Bag,
+		qma *aspect.QuotaMethodArgs) (*adapter.QuotaResult2, error)
+}
+
+// dispatchInfo contains details needed to make a function call
+type dispatchInfo struct {
+	// generated code - processor.ProcessXXX ()
+	processor *template.Info
+	// Name of the handler being called.
+	handlerName string
+	// handler to call.
+	handlerState *HandlerState
+	// instance configuration to call the handler with.
+	instances []*cpb.Instance
+}
+
+// Report dispatches to the set of adapters associated with the Report API method
+// Config validation ensures that things are consistent.
+// If they are not, we should continue as far as possible on the runtime path
+// before aborting. Returns an error if any of the adapters return an error.
+// If not the results are combined to a single CheckResult.
+// Dispatcher#Report.
+func (m *Manager2) Report(ctx context.Context, requestBag attribute.Bag) error {
+	rt := m.RuntimeState()
+	// cannot be null by construction.
+	calls, err := fetchDispatchInfo(requestBag, adptTmpl.TEMPLATE_VARIETY_REPORT, rt, m.templateRepo, nil)
+	if err != nil {
+		return err
+	}
+
+	ra := make([]*runArg, 0, 10)
```

Comment:
why 10? should this be a const somewhere (as it is also used in `Check` and `Quota`) ?